byterun/win32.c

```diff
@@ -861,3 +866,32 @@ CAMLexport caml_stat_string caml_stat_strdup_of_utf16(const wchar_t *s)
 
   return out;
 }
+
+void caml_probe_os_version(void)
+{
+  /* Determine the version of Windows we're running, and cache it */
+  WCHAR fileName[MAX_PATH];
+  DWORD size =
+    GetModuleFileName(GetModuleHandle(L"kernel32"), fileName, MAX_PATH);
+  DWORD dwHandle = 0;
+  BYTE* versionInfo;
+  fileName[size] = 0;
+  size = GetFileVersionInfoSize(fileName, &dwHandle);
+  versionInfo = (BYTE*)malloc(size * sizeof(BYTE));
+  if (GetFileVersionInfo(fileName, 0, size, versionInfo)) {
+    UINT len = 0;
+    VS_FIXEDFILEINFO* vsfi = NULL;
+    VerQueryValue(versionInfo, L"\\", (void**)&vsfi, &len);
+    caml_win32_major = HIWORD(vsfi->dwFileVersionMS);
+    caml_win32_minor = LOWORD(vsfi->dwFileVersionMS);
+    caml_win32_build = HIWORD(vsfi->dwFileVersionLS);
+    caml_win32_revision = LOWORD(vsfi->dwFileVersionLS);
+  }
+  free(versionInfo);
+}
+
+void caml_setup_terminal(void)
+{
+  if (caml_win32_major >= 10)
+    SetConsoleOutputCP(CP_UTF8);
+}
```

Comment:
Maybe you want to guard with `#ifdef WINDOWS_UNICODE`.  Or not?

Reply:
I think it's clearer for the `#if WINDOWS_UNICODE` to be sys.c? So the function always does this, but we only actually call it if `WINDOWS_UNICODE=1`